### ISSUE TITLE
Once the initialization fails, the initialization of the later always fail

### ISSUE THREAD

**Issue by kazuki** [SamuraiT/mecab-python3#35]
version: 0.996.2

How to reproduce:
```
from MeCab import Tagger
Tagger()  # ok
try:
    Tagger('-hoge')  # invalid args. throws RuntimeError
except:
    pass
Tagger()  # throws RuntimeError. why ?
```

version 0.996.1: no problem. works fine

**Comment by kazuki** [comment]
pip install version isn't work.
But build from source version is works.

Build environment:
* dist: gentoo
* gcc: 9.2.0
* swig: 3.0.12

**Comment by polm** [comment]
I managed to track down the source of this, it appears to be related to the handling of the bundled dictionary. If you look at `_mecabrc_for_bundled_dictionary` it sets the `MECABRC` environment variable to a temp directory. It's supposed to delete that setting afterwards, but when an exception is thrown it's not cleaning it up properly, so if you try to create a tagger with the env var pointing at a temp dir that no longer exists you get an error.

However, I can only reproduce this behavior with the 0.996.2 wheel (same as you) in an IPython interpreter - in a script it seems to work OK. The code hasn't changed between 0.996.2 and master so I think IPython must be doing something strange. Are you getting this behavior in a script or an interpreter?

**Comment by zackw** [comment]
Ugh, this is my error.  I always forget that the `yield` in a `@contextlib.contextmanager` will rethrow any exception that happens internally, so you have to use a try-finally block or a with clause for any cleanup that needs to happen on exceptions.  Should be fixed by d149a9d.
